packages/components/src/DateTimePickers/InputDateTimeRangePicker/InputDateTimeRangePicker.component.js

```diff
@@ -23,9 +24,56 @@ const PROPS_TO_OMIT_FOR_INPUT = [
 ];
 
 function InputDateTimeRangePicker(props) {
-	const { id, dateFormat, useSeconds, onChange, onBlur } = props;
+	const { id, dateFormat, useSeconds, onChange, onBlur, responsive } = props;
 	const inputProps = omit(props, PROPS_TO_OMIT_FOR_INPUT);
 
+	const [vertical, setVertical] = useState(false);
+	const containerRef = useRef();
+
+	const className =
+		props.vertical || vertical
+			? classnames(theme['range-picker-vertical'], 'range-picker-vertical')
+			: classnames(theme['range-picker'], 'range-picker');
+
+	function tryHorizontal() {
+		const rangeContainer = containerRef.current;
+		if (rangeContainer) {
+			const nodes = rangeContainer.querySelectorAll('input');
+			const lastChild = nodes[nodes.length - 1];
+			if (lastChild) {
+				if (
+					rangeContainer.getBoundingClientRect().right < lastChild.getBoundingClientRect().right
+				) {
+					setVertical(true);
+				}
+			}
+		} else {
+			setVertical(false);
+		}
+	}
+
+	function showHorizontalAndTest() {
+		setVertical(false);
+		tryHorizontal();
+	}
+
+	useEffect(() => {
+		if (responsive) {
+			const resizeListener = window.addEventListener(
+				'resize',
+				debounce(showHorizontalAndTest, 200),
+			);
+			return () => window.removeEventListener('resize', resizeListener);
+		}
+		return undefined;
+	}, []);
+
+	useEffect(() => {
+		if (responsive) {
+			setTimeout(() => tryHorizontal());
```

Comment:
wait until date and time input fields to be set width by InputSizer, then tryHorizontal to see if enough space for horizontal layout.